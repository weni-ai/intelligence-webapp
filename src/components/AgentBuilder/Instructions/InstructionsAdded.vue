<template>
  <section class="instructions-added">
    <h2
      class="instructions-added__title"
      data-testid="instructions-title"
    >
      {{ $t('agent_builder.instructions.instructions_added.title') }}
    </h2>

    <section
      :class="[
        'instructions-added__instructions',
        {
          'instructions-added__instructions--loading':
            instructionsStore.instructions.status === 'loading',
        },
      ]"
      data-testid="instructions"
    >
      <template v-if="instructionsStore.instructions.status === 'loading'">
        <UnnnicSkeletonLoading
          v-for="index in 10"
          :key="index"
          tag="div"
          width="100%"
          height="60px"
          data-testid="instruction-loading"
        />
      </template>
      <template v-else></template>
    </section>
  </section>
</template>

<script setup>
import { useInstructionsStore } from '@/store/Instructions';

const instructionsStore = useInstructionsStore();
instructionsStore.instructions.status = 'loading';
</script>

<style lang="scss" scoped>
.instructions-added {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__title {
    margin: 0;

    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  }

  &__instructions {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-nano;

    border-radius: $unnnic-border-radius-md;
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

    &--loading {
      border: none;
      border-radius: 0;
    }
  }
}
</style>
