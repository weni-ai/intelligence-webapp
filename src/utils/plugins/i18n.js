import * as VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import ptbr from '../../locales/pt_br.json';
import es from '../../locales/es.json';

const languages = {
  'en-US': en,
  'pt-BR': ptbr,
  es,
};

const messages = Object.assign(languages);

const i18n = VueI18n.createI18n({
  locale: 'en-US',
  fallbackLocale: 'en-US',
  messages,
  warnHtmlInMessage: 'off',
});

export default i18n;
