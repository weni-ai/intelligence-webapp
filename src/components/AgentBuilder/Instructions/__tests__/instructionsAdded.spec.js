import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';
import { createTestingPinia } from '@pinia/testing';

import InstructionsAdded from '../InstructionsAdded.vue';
import i18n from '@/utils/plugins/i18n';

describe('InstructionsAdded.vue', () => {
  let wrapper;
  const pinia = createTestingPinia({
    initialState: {
      instructions: {
        status: 'loading',
      },
    },
  });

  beforeEach(() => {
    wrapper = shallowMount(InstructionsAdded, {
      global: {
        plugins: [pinia],
      },
    });
  });

  const SELECTORS = {
    title: '[data-testid="instructions-title"]',
    instructions: '[data-testid="instructions"]',
    instructionLoading: '[data-testid="instruction-loading"]',
  };

  const findComponent = (component) =>
    wrapper.findComponent(SELECTORS[component]);
  const find = (selector) => wrapper.find(SELECTORS[selector]);

  const translation = (key) =>
    i18n.global.t(`agent_builder.instructions.instructions_added.${key}`);

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('Component rendering', () => {
    it('renders components correctly', () => {
      expect(find('title').exists()).toBe(true);
      expect(find('instructions').exists()).toBe(true);
    });

    it('renders title correctly', () => {
      expect(find('title').text()).toBe(translation('title'));
    });
  });

  describe('Loading state', () => {
    it('renders loading elements when status is loading', () => {
      expect(find('instructions').exists()).toBe(true);
      expect(find('instructions').classes()).toContain(
        'instructions-added__instructions--loading',
      );
      expect(
        wrapper.findAllComponents(SELECTORS.instructionLoading).length,
      ).toBe(10);
    });

    it('hides loading list when status is complete', async () => {
      wrapper.vm.instructionsStore.instructions.status = 'complete';
      await wrapper.vm.$nextTick();
      expect(findComponent('instructionLoading').exists()).toBe(false);
    });
  });
});
