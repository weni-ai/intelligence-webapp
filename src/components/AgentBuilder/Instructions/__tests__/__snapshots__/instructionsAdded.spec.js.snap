// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InstructionsAdded.vue > should match snapshot 1`] = `
"<section data-v-d1e6c271="" class="instructions-added">
  <h2 data-v-d1e6c271="" class="instructions-added__title" data-testid="instructions-title">Instructions added</h2>
  <section data-v-d1e6c271="" class="instructions-added__instructions instructions-added__instructions--loading" data-testid="instructions">
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
    <unnnic-skeleton-loading-stub data-v-d1e6c271="" width="100%" height="60px" count="1" duration="1.5" tag="div" circle="false" direct="false" data-testid="instruction-loading"></unnnic-skeleton-loading-stub>
  </section>
</section>"
`;
